torch/_inductor/codegen/cpp_wrapper_gpu.py

```diff
@@ -208,6 +208,27 @@ def write_get_raw_stream(self, device_idx: int, graph=None) -> str:
         )
         return name
 
+    def codegen_inputs(self):
+        # See Note: [Input Alignment handling in Inductor]
+        #
+        # JIT Inductor does not guard on input alignment. It relies on copy_misaligned_inputs to
+        # copy misaligned inputs to aligned buffers. For AOTInductor, we expect users to use it
+        # as non-Python deployment for its best performance, so implicitly copying misaligned inputs
+        # to aligned buffers is going to bring a surprising performance hit. Instead, we check input
```

Comment:
i'm not convinced it's that much of a performance hit. a single copy is usually not that slow..

Reply:
@chenyang78 , soliciting your opinion. Which one do you think makes more sense for production scenario, clone or assert?

Reply:
Off the top of my head, I don't recall any production model with misaligned inputs. I slightly prefer adding checks rather than creating clones. If we hit any assertion failures in a production model, we can conduct a real benchmark to check the performance impact of using clones.